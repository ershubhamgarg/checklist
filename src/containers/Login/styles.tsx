import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    alignItems: 'center',
    justifyContent: 'center',
  },
  btnContainer: {
    paddingHorizontal: 50,
    paddingVertical: 15,
    backgroundColor: '#cdcdcd',
    borderRadius: 5,
    shadowOpacity: 0.2,
  },
  txt: {
    fontSize: 15,
  },
});
