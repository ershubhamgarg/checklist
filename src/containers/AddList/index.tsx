import {BlurView} from '@react-native-community/blur';
import {useNavigation} from '@react-navigation/native';
import * as React from 'react';
import {Pressable, TextInput, View} from 'react-native';
import {useDispatch} from 'react-redux';
import Header from '../../components/AppHeader';
import Icon from '../../components/Icon';
import ListText from '../../components/ListText';
import {COLORS} from '../../constants/colors';
import {addNewList} from '../../store/reducers/mychecklistreducer';
import {styles} from './styles';
export function AddList() {
  const navigation = useNavigation();
  const dispatch = useDispatch();
  const [text, setText] = React.useState('');

  const onBack = () => {
    navigation.goBack();
  };

  const onPressCross = () => {};

  const onChangeText = e => {
    setText(e);
  };

  const onDone = () => {
    let newListObj = {
      title: text,
      listId: Date.now().toString(),
      personal: true,
      createdOn: Date.now(),
      items: [],
    };

    dispatch(addNewList(newListObj));
    navigation.replace('EditList', {listData: newListObj});
  };
  return (
    <BlurView
      blurAmount={0.2}
      reducedTransparencyFallbackColor="white"
      // blurRadius={1}
      style={styles.container}>
      <Header
        backButtonBackgroundStyle={styles.backButton}
        onBackPress={onBack}>
        {text && (
          <Pressable onPress={onDone} style={styles.done}>
            <ListText bold style={styles.doneText}>
              Done
            </ListText>
          </Pressable>
        )}
      </Header>
      <View style={styles.inputOuter}>
        <View style={styles.inputContainer}>
          <TextInput
            value={text}
            onChangeText={onChangeText}
            placeholder="Insert listâ€™s title"
            placeholderTextColor={COLORS.MARLOW_GREY}
            style={styles.input}
          />
          {text && (
            <Pressable onPress={onPressCross}>
              <Icon name={'cross'} />
            </Pressable>
          )}
        </View>
      </View>
    </BlurView>
  );
}
