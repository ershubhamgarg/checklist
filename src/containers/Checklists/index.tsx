import {NavigationProp, useNavigation} from '@react-navigation/native';
import * as React from 'react';
import {FlatList, Pressable, SafeAreaView} from 'react-native';
import Header from '../../components/AppHeader';
import ChecklistHeader from '../../components/ChecklistHeader';
import ChecklistItem from '../../components/ChecklistItem';
import json from './../../json/ChecklistHome.json';
import {styles} from './styles';
import Icon from '../../components/Icon';
import {ChecklistItemProps} from './type';

export function Checklists() {
  const navigation = useNavigation();
  const {navigate} = navigation;
  const goback = () => {
    navigation.goBack();
  };

  const onHeaderCardPress = () => {
    navigate('PreDepartureList');
  };

  const onListCardPress = () => {
    // alert(e.title);
  };

  const onPressAdd = () => {
    navigate('AddList');
  };

  const _renderItem = ({item, index}: ChecklistItemProps) => {
    return (
      <ChecklistItem item={item} index={index} onCardPress={onListCardPress} />
    );
  };

  const _listHeader = () => {
    return <ChecklistHeader onCardPress={onHeaderCardPress} />;
  };
  return (
    <SafeAreaView style={styles.container}>
      <Header onBackPress={goback} title={'Checklists'} />

      <FlatList
        onRefresh={() => {}}
        refreshing={!true}
        renderItem={_renderItem}
        ListHeaderComponent={_listHeader}
        showsVerticalScrollIndicator={false}
        data={json}
      />
      <Pressable onPress={onPressAdd}>
        <Icon name="add" />
      </Pressable>
    </SafeAreaView>
  );
}
