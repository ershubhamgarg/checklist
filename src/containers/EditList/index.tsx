import {useNavigation} from '@react-navigation/native';
import * as React from 'react';
import {FlatList, Keyboard, Pressable, SafeAreaView, View} from 'react-native';

import {} from 'react-native-safe-area-context';
import {useDispatch, useSelector} from 'react-redux';
import Header from '../../components/AppHeader';
import ListText from '../../components/ListText';
import MyListItem from '../../components/MyListItem';
import json from './../../json/myList.json';
import {styles} from './styles';
import {saveItemToList} from '../../store/reducers/mychecklistreducer';
export function EditList(props) {
  const {route} = props;
  const {params} = route;
  const {listData} = params;

  const navigation = useNavigation();
  const dispatch = useDispatch();

  const [editing, setEditing] = React.useState(false);
  const [text, setText] = React.useState('');
  const {myList} = useSelector(state => state.mychecklistreducer);

  const [data, setData] = React.useState([]);

  React.useEffect(() => {
    let arr = myList.filter((item: any) => item.listId === listData.listId)[0]
      .items;
    if (arr?.length) {
      setData(arr);
    } else {
      onEdit();
    }
  }, [listData, myList]);

  const onBack = () => {
    navigation.goBack();
  };

  const onEdit = () => {
    let editObj = {input: true};
    let arr = data ? data.slice() : [];
    arr.push(editObj);
    setData(arr);
    setEditing(true);
  };

  const onSave = () => {
    if (text.length) {
      let newItemObj = {
        title: text,
        listId: listData.listId,
        itemId: Date.now().toString(),
      };
      console.log('new item : ', newItemObj);
      dispatch(saveItemToList(newItemObj));
    } else {
      let arr = data;
      arr.pop();
      setData(arr);
    }
    setEditing(false);
  };

  const onChange = e => {
    setText(e);
  };

  const renderItem = ({item, index}) =>
    item?.input ? (
      <MyListItem
        index={index}
        input={true}
        item={item}
        onCardPress={() => {}}
        onChange={onChange}
      />
    ) : (
      <MyListItem
        text={text}
        index={index}
        item={item}
        onCardPress={() => {}}
      />
    );

  return (
    <SafeAreaView
      // blurRadius={1}
      style={styles.container}>
      <Header backLabel="Lists" onBackPress={onBack}>
        {editing ? (
          <Pressable onPress={onSave} style={styles.done}>
            <ListText bold style={styles.doneText}>
              Save
            </ListText>
          </Pressable>
        ) : (
          <Pressable onPress={onEdit} style={styles.edit}>
            <ListText bold style={styles.editText}>
              Edit List
            </ListText>
          </Pressable>
        )}
      </Header>
      <View style={{flex: 1}}>
        <View style={{padding: 20}}>
          <ListText bold style={{fontSize: 18}}>
            {listData.title}
          </ListText>
        </View>
        <View style={{flex: 1}}>
          <FlatList data={data} renderItem={renderItem} />
        </View>
      </View>
    </SafeAreaView>
  );
}
