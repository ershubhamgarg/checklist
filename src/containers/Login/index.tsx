import {useNavigation} from '@react-navigation/native';
import * as React from 'react';
import {Pressable, SafeAreaView, Image, View} from 'react-native';
import {styles} from './styles';
import ListText from '../../components/ListText';

export function Login() {
  const navigation = useNavigation();
  const onLoginPress = () => {
    navigation.navigate('Checklists');
  };
  return (
    <SafeAreaView style={styles.container}>
      <View>
        <Image
          style={styles.image}
          source={require('./../../assets/images/mn.jpeg')}
          resizeMode="contain"
        />
        <Pressable style={styles.btnContainer} onPress={onLoginPress}>
          <ListText medium style={styles.txt}>
            Login to Marlow checklist
          </ListText>
        </Pressable>
      </View>
    </SafeAreaView>
  );
}
