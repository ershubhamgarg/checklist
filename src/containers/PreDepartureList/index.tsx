import {useNavigation} from '@react-navigation/native';
import * as React from 'react';
import {
  FlatList,
  SafeAreaView,
  ScrollView,
  StyleSheet,
  TouchableOpacity,
  View,
} from 'react-native';
import {SceneMap, TabView} from 'react-native-tab-view';
import {useDispatch, useSelector} from 'react-redux';
import Header from '../../components/AppHeader';
import Icon from '../../components/Icon';
import {ListProgress} from '../../components/ListProgress';
import ListText from '../../components/ListText';

import PDListItem from '../../components/PDListItem';
import {COLORS} from '../../constants/colors';
import {
  doneItemFromPDList,
  skipItemFromPDList,
  submitItemFromPDList,
} from '../../store/reducers/mychecklistreducer';
import {styles} from './styles';
import {usePDListProgress} from '../../utils';
export function PreDepartureList() {
  const navigation = useNavigation();
  const [tab, setTab] = React.useState(0);

  const {pdList} = useSelector(state => state.mychecklistreducer);
  const dispatch = useDispatch();
  const goback = () => {
    navigation.goBack();
  };

  const [progress] = usePDListProgress();

  const _handleIndexChange = e => {
    setTab(e);
  };

  const onPressDone = e => {
    let listId = pdList[tab].key;
    let obj = {...e, listId: listId};

    dispatch(doneItemFromPDList(obj));
  };

  const onPressSubmitted = e => {
    let listId = pdList[tab].key;
    let obj = {...e, listId: listId};

    dispatch(submitItemFromPDList(obj));
  };
  const onPressSkipped = e => {
    let listId = pdList[tab].key;
    let obj = {...e, listId: listId};

    dispatch(skipItemFromPDList(obj));
  };

  const _renderItem = ({item, index}) =>
    // : ChecklistItemProps
    {
      return (
        <PDListItem
          item={item}
          index={index}
          onPressDone={onPressDone}
          onPressSubmitted={onPressSubmitted}
          onPressSkipped={onPressSkipped}
        />
      );
    };

  const FirstRoute = ({...e}) => {
    let data = e.route.items;
    let completed = data.filter(e => e.completed || e.skipped);
    let pending = data.filter(e => !e.completed);
    return (
      <ScrollView
        style={[styles2.container, {backgroundColor: COLORS.MARLOW_NAVY}]}>
        <View style={{padding: 25}}>
          <ListText italic style={{fontSize: 12}}>
            Items should only be ticked off once the corresponding original
            paper document has been added to your Blue Pouch in preparation for
            departure.
          </ListText>
        </View>
        {pending.length ? (
          <ListText bold style={styles.label}>
            Pending
          </ListText>
        ) : null}
        {pending.map((item, index) => _renderItem({item, index}))}

        {completed.length ? (
          <ListText bold style={styles.label}>
            Completed
          </ListText>
        ) : null}
        {completed.map((item, index) => _renderItem({item, index}))}
      </ScrollView>
    );
  };
  const SecondRoute = e => {
    return (
      <View
        style={[styles2.container, {backgroundColor: COLORS.MARLOW_NAVY}]}
      />
    );
  };
  const _renderTabBar = props => {
    const {navigationState} = props;

    return (
      <View>
        <FlatList
          bounces={false}
          showsHorizontalScrollIndicator={false}
          renderItem={({item, index}) => {
            return (
              <View style={{}}>
                <TouchableOpacity
                  onPress={() => setTab(index)}
                  style={[
                    styles2.tabItem,
                    {
                      flexDirection: 'row',
                      borderBottomColor: COLORS.MARLOW_BLUE,
                      borderBottomWidth: tab === index ? 2 : 0,
                    },
                  ]}>
                  {item.requireAttention && <Icon name={'attention'} />}
                  <ListText
                    bold
                    style={{
                      color:
                        tab === index
                          ? COLORS.MARLOW_BLUE
                          : COLORS.MARLOW_NAVY_500,
                    }}>
                    {item.title}
                  </ListText>
                </TouchableOpacity>
              </View>
            );
          }}
          horizontal
          data={navigationState?.routes}
        />
      </View>
    );
  };

  const _renderScene = SceneMap({
    first: FirstRoute,
    second: SecondRoute,
    third: SecondRoute,
    fourth: SecondRoute,
    fifth: SecondRoute,
  });

  return (
    <SafeAreaView style={styles.container}>
      <Header
        backLabel={'Back'}
        onBackPress={goback}
        title={'Pre-Departure Documents List'}></Header>
      <View style={{paddingVertical: 20}}>
        <ListProgress progress={progress} />
      </View>
      <TabView
        // swipeEnabled={false}
        navigationState={{
          index: tab,
          routes: pdList,
        }}
        renderScene={_renderScene}
        renderTabBar={_renderTabBar}
        onIndexChange={_handleIndexChange}
      />
    </SafeAreaView>
  );
}

const styles2 = StyleSheet.create({
  container: {
    flex: 1,
    height: 1100,
  },
  tabBar: {
    flexDirection: 'row',
  },
  tabItem: {
    alignItems: 'center',
    padding: 10,
  },
});
