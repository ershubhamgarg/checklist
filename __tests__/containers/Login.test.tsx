import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Login} from '../../src/containers/Login';

describe('Login screen', () => {
  const tree = renderer.create(<Login />);
  test('Login screen snapshot created successfully', () => {
    expect(tree).toMatchSnapshot();
  });

  test('Login screen rendered successfully', () => {
    expect(tree).toBeTruthy();
  });
});
