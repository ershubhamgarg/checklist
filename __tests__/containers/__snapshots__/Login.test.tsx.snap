// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen Login screen snapshot created successfully 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Image
      resizeMode="contain"
      source={
        {
          "testUri": "../../../src/assets/images/mn.jpeg",
        }
      }
      style={
        {
          "alignSelf": "center",
          "height": 300,
          "width": 300,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#005AA5",
          "borderRadius": 25,
          "borderWidth": 2,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "Roboto-Medium",
            },
            {
              "color": "#005AA5",
              "fontSize": 15,
            },
          ]
        }
      >
        Go to checklist
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
